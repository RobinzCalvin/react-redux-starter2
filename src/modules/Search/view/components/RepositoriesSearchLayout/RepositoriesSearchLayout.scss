.repositories-search-layout {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  min-height: 77vh; // TODO: возможно сделать элегантнее

  // TODO: в UsersSearch такие же стили

  &__search-form {
    margin: 0.8rem;
    width: 30rem;

    @media (max-width: 490px) {
      width: 95%;
    }
  }

  &__pagination {
    margin-top: 1.5rem;
  }
}
