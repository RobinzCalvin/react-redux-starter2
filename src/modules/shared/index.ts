export { default as Layout } from './Layout/Layout';
export { default as withLayout } from './withLayout';
