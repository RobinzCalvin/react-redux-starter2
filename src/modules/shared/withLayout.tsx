import React from 'react';

import Layout from './Layout/Layout';

function withLayout(title: string) {
  return (Component: React.ComponentType) => () => (
    <Layout title={title}>
      <Component />
    </Layout>
  );
}

export default withLayout;
