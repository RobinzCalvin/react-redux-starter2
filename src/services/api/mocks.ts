import { ICakePreview } from 'shared/types/models';

// tslint:disable:max-line-length
export const cakesPreview: ICakePreview[] =  [
  {
    name: 'Vanilla Almond',
    description: 'Light and sweet but moist and rich in flavor, our Vanilla Almond cake is filled and frosted with our house vanilla buttercream. Vanilla Almond Amycakes are decorated with sparkly clear crystal sprinkles, pearl sprinkles and a ring of buttercream rosettes',
    imageURL: '',
  },
  {
    name: 'Golden Vanilla',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house vanilla buttercream. Golden Vanilla Amycakes are decorated with ascending gold sprinkles, white ganache drizzle and white chocolate curls and flakes',
    imageURL: '',
  },
  {
    name: 'Golden Chocolate',
    description: 'Old fashioned, buttery golden yellow cake, just as moist and dense as Grandma used to make. Filled and frosted with our house chocolate buttercream. Golden Chocolate Amycakes are decorated with ascending gold sprinkles, chocolate ganache drizzle and dark chocolate curls and flakes.',
    imageURL: '',
  },
  {
    name: 'Confetti',
    description: 'Moist Vanilla Almond cake studded with confetti sprinkles in the batter. Confetti Amycakes are frosted with vanilla buttercream and decorated with colorful sprinkles.',
    imageURL: '',
  },
  {
    name: 'Zebra',
    description: 'Alternating layers of vanilla almond and chocolate cakes, filled and frosted with our house vanilla buttercream.  Zebra Amycakes are decorated with chocolate ganache drizzle and white and chocolate curls. ',
    imageURL: '',
  },
];
