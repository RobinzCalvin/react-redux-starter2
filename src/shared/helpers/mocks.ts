import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export function createContainerMock() {
  return function ContainerMock() {
    return null;
  } as any;
}

export function getProfileMock(): IProfile {
  return {
    age: 100,
    nickname: 'Nickname',
    name: 'Name name',
    bio: 'profile bio',
    avatarURL: 'https://the-url.com',
  };
}

export function getGithubUserMock(): IGithubUser {
  return {
    id: 1,
    username: 'username',
    avatarURL: 'http://the-url.com',
    htmlURL: 'http://the-url.com',
  };
}

export function getDetailedGithubUserMock(): IDetailedGithubUser {
  return {
    ...getGithubUserMock(),
    followersNumber: 12,
    followingNumber: 12,
    reposNumber: 12,
    realName: 'The real name',
    location: 'Nowhere',
  };
}

export function getRepositoryMock(): IRepository {
  return {
    id: 1,
    openIssuesNumber: 1,
    starsNumber: 1,
    forksNumber: 1,
    description: 'description',
    name: 'name',
    htmlURL: 'http://the-url.com',
    language: 'en',
    updatedAt: '05/05/2005',
    owner: getGithubUserMock(),
  };
}
// TODO: моки?
