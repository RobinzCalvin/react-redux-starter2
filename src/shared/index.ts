const testContext = (require as any).context('./', true, /^.*\/(?!.*test).*\.tsx?$/);
testContext.keys().forEach(testContext);
