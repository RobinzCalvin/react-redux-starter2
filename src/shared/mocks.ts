import { IProfile, IGithubUser, IDetailedGithubUser, IRepository } from 'shared/types/models';

export const Container: any = () => null;

export const profile: IProfile = {
  age: 100,
  nickname: 'Nickname',
  name: 'Name name',
  bio: 'profile bio',
  avatarURL: 'https://the-url.com',
};

export const githubUser: IGithubUser = {
  id: 1,
  username: 'username',
  avatarURL: 'http://the-url.com',
  htmlURL: 'http://the-url.com',
};

export const detailedGithubUser: IDetailedGithubUser = {
  ...githubUser,
  followersNumber: 12,
  followingNumber: 12,
  reposNumber: 12,
  realName: 'The real name',
  location: 'Nowhere',
};

export const repository: IRepository = {
  id: 1,
  openIssuesNumber: 1,
  starsNumber: 1,
  forksNumber: 1,
  description: 'description',
  name: 'name',
  htmlURL: 'http://the-url.com',
  language: 'en',
  updatedAt: '05/05/2005',
  owner: githubUser,
};
