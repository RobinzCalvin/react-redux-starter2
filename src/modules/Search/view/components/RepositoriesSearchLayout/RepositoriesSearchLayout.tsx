import React, { useState } from 'react';
import block from 'bem-cn';

import * as features from 'features';
import featureConnect from 'core/FeatureConnector';

import Layout from 'modules/shared/Layout/Layout';
import './RepositoriesSearchLayout.scss';

interface IFeatureProps {
  repositoriesSearchFeatureEntry: features.repositoriesSearch.Entry;
}

type IProps = IFeatureProps;

const b = block('repositories-search-layout');

function RepositoriesSearchLayout(props: IProps) {
  const { repositoriesSearchFeatureEntry: { containers } } = props;
  const { RepositoriesSearchForm, RepositoriesSearchResults, RepositoriesSearchPagination } = containers;
  const [lastSubmittedFormState, setLastSubmittedFormState] =
    useState<features.repositoriesSearch.namespace.IRepositoriesSearchFormFields | null>(null);

  return (
    <Layout title="GitHub repositories search">
      <div className={b()}>
        <div className={b('search-form')}>
          <RepositoriesSearchForm onSubmit={setLastSubmittedFormState}/>
        </div>
        <div className={b('results')}>
          <RepositoriesSearchResults />
        </div>
        {lastSubmittedFormState !== null && // рендерить когда есть резалты?
          <div className={b('pagination')}>
            <RepositoriesSearchPagination formFields={lastSubmittedFormState} />
          </div>
        }
      </div>
    </Layout>
  );
}

export default featureConnect({
  repositoriesSearchFeatureEntry: features.repositoriesSearch.loadEntry,
})(RepositoriesSearchLayout);
