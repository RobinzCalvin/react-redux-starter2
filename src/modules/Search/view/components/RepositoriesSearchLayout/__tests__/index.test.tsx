import React from 'react';
import { render } from 'enzyme';

import { makeMockEntry } from 'shared/mocks';

import { RepositoriesSearchLayout, IRepositoriesSearchLayoutProps } from '../RepositoriesSearchLayout';

const props: IRepositoriesSearchLayoutProps = {
  repositoriesSearchFeatureEntry: makeMockEntry(),
};

describe('RepositoriesSearchLayout component', () => {
  it('should match snapshot', () => {
    const component = render(<RepositoriesSearchLayout {...props} />);
    expect(component).toMatchSnapshot();
  });
});
