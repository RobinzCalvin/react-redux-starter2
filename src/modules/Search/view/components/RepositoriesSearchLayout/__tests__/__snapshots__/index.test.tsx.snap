// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesSearchLayout component should match snapshot 1`] = `null`;
